lib/MusicBrainz/Server/Controller/WS/2/Genre.pm

```diff
@@ -0,0 +1,70 @@
+package MusicBrainz::Server::Controller::WS::2::Genre;
+use Moose;
+use MusicBrainz::Server::Validation qw( is_guid );
+
+BEGIN { extends 'MusicBrainz::Server::ControllerBase::WS::2' }
+
+use aliased 'MusicBrainz::Server::WebService::WebServiceStash';
+
+my $ws_defs = Data::OptList::mkopt([
+    genre => {
+        method   => 'GET',
+        required => [ qw(query) ],
+        optional => [ qw(fmt limit offset) ],
+    },
+    genre => {
+        method   => 'GET',
+        optional => [ qw(fmt limit offset) ],
+    },
+    genre => {
+        method   => 'GET',
+        optional => [ qw(fmt) ],
+    }
+]);
+
+with 'MusicBrainz::Server::WebService::Validator' => {
+    defs => $ws_defs,
+};
+
+with 'MusicBrainz::Server::Controller::WS::2::Role::Lookup' => {
+    model => 'Genre',
+};
+
+sub base : Chained('root') PathPart('genre') CaptureArgs(0) { }
+
+sub genre_toplevel {
+    my ($self, $c, $stash, $genres) = @_;
+
+}
+
+sub genre_browse : Private {
```

Comment:
Given the previous comment I'm guessing browsing isn't working yet (or I'm not sure how to use it). I'd expect, I guess, to be able to get genres for the Beatles via `/ws/2/genre/?artist=b10bbbfc-cf9e-42e0-be17-e2c3e1d2600d`? But that gives me `Invalid resource genre.`.

So we should either finish the implementation or remove the broken bits and leave them for a later PR (returning a 501 for now, I guess, like with search).

Reply:
Yes, only lookup is implemented, neither browse nor search.

Reply:
I'm not sure browse makes sense, honestly. Do you feel it should be added?

Reply:
Not sure - I know you'd normally get them via just inc=genres or user-genres on the entity request, but if we were to add things like aliases and relationships, those probably wouldn't be included there, and a browse request might make sense. Though it's not something we have to include in this PR (I'd just remove the broken code for now).

Reply:
I'm returning 501 for now.